src/rendering/framebufferrenderer.cpp

```diff
@@ -1206,169 +1194,148 @@ void FramebufferRenderer::render(float blackoutFactor, bool doPerformanceMeasure
     }
-}
 
-float FramebufferRenderer::hdrBackground() const {
-    return _hdrBackground;
-}
+    void saveTextureToPPMFile(const GLenum color_buffer_attachment,
+        const std::string & fileName, const int width, const int height)
+    {
+        std::fstream ppmFile;
 
-int FramebufferRenderer::nAaSamples() const {
-    return _nAaSamples;
-}
+        ppmFile.open(fileName.c_str(), std::fstream::out);
+        if (ppmFile.is_open()) {
+            unsigned char* pixels = new unsigned char[width*height * 3];
+            for (int t = 0; t < width*height * 3; ++t) {
+                pixels[t] = 255;
+            }
 
-std::vector<double> FramebufferRenderer::mSSAPattern() const {
-    return _mSAAPattern;
-}
+            if (color_buffer_attachment != GL_DEPTH_ATTACHMENT) {
+                glReadBuffer(color_buffer_attachment);
+                glReadPixels(0, 0, width, height, GL_RGB, GL_UNSIGNED_BYTE, pixels);
 
-void FramebufferRenderer::updateRendererData() {
-    ghoul::Dictionary dict;
-    dict.setValue("fragmentRendererPath", absPath(RenderFragmentShaderPath));
+            }
+            else {
+                glReadPixels(
+                    0,
+                    0,
+                    width,
+                    height,
+                    GL_DEPTH_COMPONENT,
+                    GL_UNSIGNED_BYTE,
+                    pixels
+                );
+            }
+
+            ppmFile << "P3" << std::endl;
+            ppmFile << width << " " << height << std::endl;
+            ppmFile << "255" << std::endl;
 
-    _rendererData = dict;
+            std::cout << "\n\nFILE\n\n";
+            int k = 0;
+            for (int i = 0; i < width; i++) {
+                for (int j = 0; j < height; j++) {
+                    ppmFile << static_cast<unsigned int>(pixels[k]) << " "
+                        << static_cast<unsigned int>(pixels[k + 1]) << " "
+                        << static_cast<unsigned int>(pixels[k + 2]) << " ";
+                    k += 3;
+                }
+                ppmFile << std::endl;
+            }
+            delete[] pixels;
 
-    OsEng.renderEngine().setRendererData(dict);
-}
+            ppmFile.close();
+        }
+    }
 
-void saveTextureToPPMFile(const GLenum color_buffer_attachment,
-                          const std::string & fileName, const int width, const int height)
-{
-    std::fstream ppmFile;
+    void saveTextureToMemory(const GLenum color_buffer_attachment,
+        const int width, const int height, std::vector<double> & memory) {
 
-    ppmFile.open(fileName.c_str(), std::fstream::out);
-    if (ppmFile.is_open()) {
-        unsigned char* pixels = new unsigned char[width*height * 3];
-        for (int t = 0; t < width*height * 3; ++t) {
-            pixels[t] = 255;
+        if (!memory.empty()) {
+            memory.clear();
         }
+        memory.resize(width * height * 3);
+
+        float *tempMemory = new float[width*height * 3];
```

Comment:
Pointer belongs to `float`, not to the variable

Better to use `std::vector<float>` anyway